.github/workflows/upstream-sync.yml

```diff
@@ -3,86 +3,84 @@ name: Upstream Sync Workflow
-              return result.data['sha'];
-            }
-          result-encoding: string
-      - name: Create copy of upstream branch
-        uses: actions/github-script@9ac08808f993958e9de277fe43a64532a609130e
-        id: branch-name
-        with:
-          script: |
-            if(${{ steps.check-if-ahead.outputs.result }} == true) {
-              const { repo, owner } = context.repo;
-              const date = new Date();
-              const prDate = `${date.getFullYear()}-${date.getMonth()}-${date.getDay()}`;
-              const branchName = '${{ format('upstream-{0}-', inputs.release-series) }}' + prDate;
-              const result = await github.rest.git.createRef({
-                owner,
-                repo,
-                ref: 'refs/heads/' + branchName,
-                sha: '${{ steps.get-upstream-sha.outputs.result }}',
-              });
-              return branchName;
-            }
-          result-encoding: string
-      - name: Create PR ${{ format('{0}/{1}', env.downstream-prefix, inputs.release-series) }} with ${{ steps.branch-name.outputs.result }}
-        uses: actions/github-script@9ac08808f993958e9de277fe43a64532a609130e
-        with:
-          script: |
-            if(${{ steps.check-if-ahead.outputs.result }} == true) {
-              const { repo, owner } = context.repo;
-              const result = await github.rest.pulls.create({
-                title: 'Synchronise ${{ inputs.release-series }} with ${{ steps.branch-name.outputs.result }}',
-                owner,
-                repo,
-                head: '${{ steps.branch-name.outputs.result }}',
-                base: '${{ format('{0}/{1}', env.downstream-prefix, inputs.release-series) }}',
-                maintainer_can_modify: false,
-                body: [
-                  'The changes from ${{ env.upstream-prefix }}/${{ inputs.release-series }} have been included in this automated PR'
-                ].join('\n')
-              });
-              github.rest.issues.addLabels({
-                owner,
-                repo,
-                issue_number: result.data.number,
-                labels: ['automated', 'sync']
-              });
-            }
+        id: get_upstream_sha
+        run: |
+          upstream_sha=$(gh api \
+          -H "Accept: application/vnd.github.v3+json" \
+          "/repos/${{env.UPSTREAM_OWNER}}/$(basename $(pwd))/commits/stable/${{inputs.release_series}}" | jq ".sha")
+          echo "::set-output name=result::$upstream_sha"
+      - name: Check if downstream branch exists
+        id: check_if_downstream_branch_exists
+        run: |
+          api_path="/repos/${{env.DOWNSTREAM_OWNER}}/$(basename $(pwd))/branches/stackhpc/${{inputs.release_series}}"
+          response=$(gh api -H "Accept: application/vnd.github.v3+json" $api_path --silent --include) || true
+          if [[ "$response" =~ "HTTP/2.0 200 OK" ]]; then
+            echo "::set-output name=result::true"
+          else
```

Comment:
Do you need to capture three cases explicitly here?

200 OK == result::true
Branch not found (HTTP 404) == result::false
Anything else == bail out because behaviour undefined